src/App.tsx

```diff
@@ -14,11 +14,93 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const [listVisible, setListVisible] = useState(false);
+  const [visibleGoods, setVisibleGoods] = useState([...goodsFromServer]);
+
+  const findIndexOfName = (currentName: string): number => (
+    goodsFromServer.findIndex(name => name === currentName)
+  );
+
+  return (
+    <div className="conteiner">
+      <div className="app">
+        {!listVisible && (
+          <button
+            className="button is-danger is-large"
+            type="button"
+            onClick={() => setListVisible(!listVisible)}
+          >
+            Start
+          </button>
+        )}
+
+        {listVisible && (
+          <div className="app__listConteiner">
+            <h1 className="app__title">Goods:</h1>
+
+            <ul
+              className="app__list"
+            >
+              {visibleGoods.map(good => (
+                <li
+                  className="app__list-item"
+                  key={good}
+                >
+                  {`${findIndexOfName(good) + 1} - ${good}`}
+                </li>
+              ))}
+            </ul>
+
+            <button
+              className="button is-primary"
+              type="button"
+              onClick={() => (setVisibleGoods(
+                [...visibleGoods].reverse(),
+              ))}
```

Comment:
If you make some changes depending on the value of previous state, you should use function inside setVisibleGoods method (as we used functional setState)
setVisibleGoods((currentGoods) => ({...}))

Reply:
You're absolutely right! Fixed.